app/components/dropdown-widget/component.js

```diff
@@ -172,47 +177,64 @@ export default Ember.Component.extend({
 
             let filter_data = await Ember.$.ajax({
-                url: 'https://dev-labs.cos.io/api/records/search/creativeworks/_search?request_cache=true',
+                url: 'https://dev-labs.cos.io/records/_search?request_cache=true',
                 crossDomain: true,
                 data: JSON.stringify(filter_query),
                 type: 'POST',
                 contentType: 'application/json'
             });
-            let afilteredList = filter_data.hits.hits.map(function(x) {
-                if (widget_category === "contributors") {
+
+            this.send('filterInput' ,widget_category, filter_data , search_term)
+
 
 
-                    let contributorsList = x._source.lists.contributors.map(function(y) {
+    
+    },
+    filterInput(type, filter_data, search_term){
+        let afilteredList = filter_data.hits.hits.map(function(x) {
+                let contributorsList;
+                
+                if(type === 'tags'){
+                    contributorsList = x._source.tags.map(function(y) {
+                        return {
+                            key: y,
+                            name: y
+                        };
+                    });
+                }else{
+                    contributorsList = x._source.lists[type].map(function(y) {
                         return {
                             key: y.id,
                             name: y.name
                         };
                     });
+                }
 
-                    var filteredContribList = contributorsList.filter(function(word) {
-                        return word.name.toLowerCase().match(search_term.toLowerCase());
-                    });
 
-                }
-                return filteredContribList;
-            });
+                var filteredContribList = contributorsList.filter(function(word) {
+                    return word.name.toLowerCase().match(search_term.toLowerCase());
+                });
 
-            let flattenedFilteredContribList = afilteredList.reduce(function(a, b) {
-              return a.concat(b);
-          }, []);
 
+               return filteredContribList;
+           });
 
 
-            for(let i = 0; i < flattenedFilteredContribList.length; i++){
-                for(let k = i+1; k < flattenedFilteredContribList.length; k++){
-                    if(flattenedFilteredContribList[i].name == flattenedFilteredContribList[k].name){
-                        flattenedFilteredContribList.splice( k, 1 );
-                    }
+
+        let flattenedFilteredContribList = afilteredList.reduce(function(a, b) {
+          return a.concat(b);
+      }, []);
+
+
+
+        for(let i = 0; i < flattenedFilteredContribList.length; i++){
+            for(let k = i+1; k < flattenedFilteredContribList.length; k++){
+                if(flattenedFilteredContribList[i].name == flattenedFilteredContribList[k].name){
+                    flattenedFilteredContribList.splice( k, 1 );
```

Comment:
we could try something like 

`flatlist.filter(contribToTest => contribToTest === flatlist.find(testContrib => testContrib.name === contribToTest.name))`

Reply:
flatlist.find returns the first obj with the same name. If contribToTest is not the same obj (ie, it wasnt the first one in flatlist), it should return false, which filters that obj out of the arr returned by flatlist.filter

Reply:
this is cool and good to know